dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java

```diff
@@ -149,9 +152,22 @@ public void executeStartUpTasks() throws DotDataException {
 
         
     }
-    
-    
-    
+
+    /**
+     * Returns the id part of a task name
+     * @param taskName
+     * @return
+     */
+    @VisibleForTesting
+    String getTaskId(final String taskName){
+        final Pattern pattern = Pattern.compile("[0-9]+");
```

Comment:
it is usually a good idea to have the pattern Compile declared static as it is immutable. and therefore thread-safe.
The matcher has to be recreated every time.  So I would  move final Pattern pattern = Pattern.compile("[0-9]+"); to a static constant.